components/brave_wallet_ui/components/desktop/popup-modals/sell-asset-modal/sell-asset-modal.tsx

```diff
@@ -76,74 +78,22 @@ export const SellAssetModal = (props: Props) => {
     return withPlaceholderIcon(AssetIcon, { size: 'medium', marginLeft: 4, marginRight: 8 })
   }, [])
 
-  const fiatBalance = React.useMemo(() => {
-    return computeFiatAmount(
-      spotPrices,
-      {
-        decimals: selectedAsset?.decimals ?? '',
-        symbol: selectedAsset?.symbol ?? '',
-        contractAddress: selectedAsset?.contractAddress ?? '',
-        value: sellAssetBalance,
-        chainId: selectedAsset?.chainId ?? ''
-      })
-  }, [
-    spotPrices,
-    sellAssetBalance,
-    selectedAsset?.symbol,
-    selectedAsset?.decimals,
-    selectedAsset?.contractAddress,
-    selectedAsset.chainId
-  ])
-
-  const estimatedAssetAmount = React.useMemo(() => {
-    if (sellAmount !== '') {
-      return `~${computeFiatAmountToAssetValue(
-        sellAmount,
-        spotPrices,
-        selectedAsset?.symbol ?? '',
-        selectedAsset?.contractAddress ?? '',
-        selectedAsset?.chainId ?? ''
-      )
-        .formatAsAsset(6, selectedAsset.symbol)}`
-    }
-    return `0.00 ${selectedAsset?.symbol}`
-  }, [
-    sellAmount,
-    spotPrices,
-    selectedAsset?.symbol,
-    selectedAsset?.contractAddress,
-    selectedAsset?.chainId
-  ])
+  const insufficientBalance = React.useMemo((): boolean => {
+    const amountWei = new Amount(sellAmount).multiplyByDecimals(
+      selectedAsset.decimals
+    )
 
-  const formattedFiatBalance = React.useMemo(() => {
-    return fiatBalance ? new Amount(fiatBalance.format(2)).formatAsFiat(defaultCurrencies.fiat) : undefined
-  }, [defaultCurrencies.fiat, fiatBalance])
-
-  const hasTooManyDecimals = React.useMemo(() => {
-    if (sellAmount !== '') {
-      return sellAmount.split('.')[1]?.length > 2
+    if (amountWei.isZero()) {
```

Comment:
Is this check necessary? We can return `amountWei.gt(sellAssetBalance)`

Reply:
Thanks for the catch! Simplified this and moved as computed.